src/block_storage/volumes.rs

```diff
@@ -0,0 +1,453 @@
+        attachments: ref Vec<protocol::VolumeAttachment>
+    }
+
+    transparent_property! {
+        #[doc = "Volume links."]
+        links: ref Vec<protocol::Link>
+    }
+
+    transparent_property! {
+        #[doc = "Name of the availability zone."]
+        availability_zone: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Current backend of the volume."]
+        host: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is encrypted."]
+        encrypted: ref bool
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the encryption key."]
+        encryption_key_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "When the volume was last updated."]
+        updated_at: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Volume replication status."]
+        replication_status: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the snapshot the volume originated from."]
+        snapshot_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume."]
+        id: ref String
+    }
+
+    transparent_property! {
+        #[doc = "Size of the volume in GiB."]
+        size: ref u64
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the user."]
+        user_id: ref String
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the project."]
+        tenant_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Migration status."]
+        migstat: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Metadata of the volume."]
+        metadata: ref HashMap<String, String>
+    }
+
+    transparent_property! {
+        #[doc = "Status of the volume."]
+        status: ref protocol::VolumeStatus
+    }
+
+    transparent_property! {
+        #[doc = "Metadata of the image used to create the volume."]
+        image_metadata: ref Option<HashMap<String, String>>
+    }
+
+    transparent_property! {
+        #[doc = "Description of the volume."]
+        description: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is multi-attachable."]
+        multi_attachable: ref bool
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume this one originated from."]
+        source_volume_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the consistency group."]
+        consistency_group_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "UUID of the volume that this volume name on the backend is based on."]
+        name_id: ref Option<String>
+    }
+
+    transparent_property! {
+        #[doc = "Name of the volume."]
+        name: ref String
+    }
+
+    transparent_property! {
+        #[doc = "Whether the volume is bootable."]
+        bootable: ref String
```

Comment:
Hmm, is it actually a string? From the name and description I'd expect a boolean...

Reply:
We could probably something similar to `Server.has_config_drive` and parse this into a `bool`. I wasn't sure here, because the cinder documentation is a little fuzzy.

The docs for the volume listing state: "Enables or disables the bootable attribute. You can boot an instance from a bootable volume." ... so this actually seems to be copy pasted from some other part of the docs.

In constract the nova docs on `config_drive` are pretty clear: "Indicates whether or not a config drive was used for this server. The value is True or an empty string. An empty string stands for False."

Reply:
I changed the `bootable` field to `bool` with the help of a new `bool_from_bootable_string` deserialization method.